servers/src/test/java/tachyon/worker/block/BlockStoreMetaTest.java

```diff
@@ -0,0 +1,194 @@
+  
+  @Before
+  public void before() throws Exception {
+    TachyonConf tachyonConf = new TachyonConf();
+    // Setup a two-tier storage: MEM:{1000}, HDD{3000, 5000}, 
+    // where missed SSD tier's quota is zero
+    String mTachyonHome = mTestFolder.newFolder().getAbsolutePath();
+    tachyonConf.set(Constants.TACHYON_HOME, mTachyonHome);
+    tachyonConf.set(Constants.WORKER_MAX_TIERED_STORAGE_LEVEL, "2");
+    tachyonConf.set(String.format(Constants.WORKER_TIERED_STORAGE_LEVEL_ALIAS_FORMAT, 0), "MEM");
+    tachyonConf.set(String.format(Constants.WORKER_TIERED_STORAGE_LEVEL_DIRS_PATH_FORMAT, 0),
+        mTachyonHome + "/ramdisk");
+    tachyonConf.set(String.format(Constants.WORKER_TIERED_STORAGE_LEVEL_DIRS_QUOTA_FORMAT, 0),
+        2000 + "");
+    tachyonConf.set(String.format(Constants.WORKER_TIERED_STORAGE_LEVEL_ALIAS_FORMAT, 1), "HDD");
+    tachyonConf.set(String.format(Constants.WORKER_TIERED_STORAGE_LEVEL_DIRS_PATH_FORMAT, 1),
+        mTachyonHome + "/disk1," + mTachyonHome + "/disk2");
+    tachyonConf.set(String.format(Constants.WORKER_TIERED_STORAGE_LEVEL_DIRS_QUOTA_FORMAT, 1),
+        3000 + "," + 5000);
+    
+    mMetadataManager = BlockMetadataManager.newBlockMetadataManager(tachyonConf);
+    
+    // Add and commit COMMITTED_BLOCKS_NUM temp blocks repeatedly
+    StorageDir dir = mMetadataManager.getTier(1).getDir(0);
+    for (long blockId = 0L; blockId < COMMITTED_BLOCKS_NUM; blockId ++) {
+      // prepare temp block
+      TempBlockMeta tempBlockMeta = new TempBlockMeta(TEST_USER_ID, blockId, TEST_BLOCK_SIZE, dir);
+      mMetadataManager.addTempBlockMeta(tempBlockMeta);
+
+      // write data
+      File tempFile = new File(tempBlockMeta.getPath());
+      
+      // create storage directory for user TEST_USER_ID at first loop
+      if (blockId == 0L) {
+        if (!tempFile.getParentFile().mkdir()) {
+          throw new IOException(String.format(
+              "Parent directory of %s can not be created for temp block", tempBlockMeta.getPath()));
+        }
+      }
+      
+      BlockWriter writer = new LocalFileBlockWriter(tempBlockMeta);
+      writer.append(BufferUtils.getIncreasingByteBuffer(Ints.checkedCast(TEST_BLOCK_SIZE)));
+      writer.close();
+
+      // commit block
+      Files.move(tempFile, new File(tempBlockMeta.getCommitPath()));
+      mMetadataManager.commitTempBlockMeta(tempBlockMeta);
+      
+    }
+    mBlockStoreMeta = new BlockStoreMeta(mMetadataManager);
+  }
+  
+  @Test
+  public void getBlockListTest() {
+    Map<Long, List<Long>> dirIdToBlockIds = new HashMap<Long, List<Long>>();
+    for (StorageTier tier : mMetadataManager.getTiers()) {
+      for (StorageDir dir : tier.getStorageDirs()) {
+        dirIdToBlockIds.put(dir.getStorageDirId(), dir.getBlockIds());
+      }
+    }
+    Map<Long, List<Long>> actual = mBlockStoreMeta.getBlockList();
+    Assert.assertEquals(dirIdToBlockIds, actual);
+    Assert.assertTrue(TEST_DIR_NUM == actual.keySet().size());
```

Comment:
Okay, i will take that method. If the program is absolutely correct, these two won't be different. Adding the latter is just for additional verification in case any mistakes from `StorageTier` & `StorageDir` which construct this test, though seem redundant.